torch/_dynamo/polyfill.py

```diff
@@ -8,3 +8,11 @@ def all(iterator):
         if not elem:
             return False
     return True
+
+
+def index(iterator, item, start=0, end=-1):
+    for i, elem in enumerate(list(iterator))[start:end]:
```

Comment:
is it really necessary to coerce to list? for e.g. large slice this could be large allocation. We do need it to slice [start:end] so maybe only coerce it in that case?

Reply:
Actually, slice doesnt have index.